tests/test_cases/test_cocotb/test_handle.py

```diff
@@ -65,3 +65,36 @@ async def test_delayed_assignment_still_errors(dut):
         dut.stream_in_int <= "1010 not a real binary string"
     with assert_raises(TypeError):
         dut.stream_in_int <= []
+
+
+@cocotb.test()
+async def test_int_vector_36bit(dut):
+    """
+    Test integer access to 36-bit vector
+    """
+    N = len(dut.stream_in_data_36bit)
+    U_MIN = 0
+    U_MAX = 2**N-1
+    S_MIN = -2**(N-1)
+    S_MAX = 2**(N-1)-1
+
+    for value in [0, 1, -1, 4, -4, U_MIN, U_MAX, S_MIN, S_MAX]:
+        dut.stream_in_data_36bit = value
+        await cocotb.triggers.Timer(10, 'ns')
+        got = int(dut.stream_in_data_36bit)
+        if value < 0:
+            got = (got & (2**(N-1)-1)) - (got & (2**(N-1)))
```

Comment:
```suggestion
        got = dut.stream_in_data_36bit.value.signed_integer
```

Reply:
This is not the same as we test both signed and unsigned values. It is a matter of choice if we interpret the bits we read back as signed or unsigned. (Eg. -1 signed is the same as max value unsigned)